certbot/compat/filesystem.py

```diff
@@ -36,6 +43,58 @@ def chmod(file_path, mode):
         _apply_win_mode(file_path, mode)
 
 
+def makedirs(file_path, mode=0o777):
+    # type: (str, int) -> None
+    """
+    Rewrite of original os.makedirs function, that will ensure on Windows that given mode
+    is correctly applied.
+    :param str file_path: The file path to open
+    :param int mode: POSIX mode to apply on leaf directory when created, Python defaults
+                     will be applied if ``None``
+    """
+    if POSIX_MODE:
+        return os.makedirs(file_path, mode)
+
+    orig_mkdir_fn = os.mkdir
+    try:
+        # As we know that os.mkdir is called internally by os.makedirs, we will swap the function in
+        # os module for the time of makedirs execution on Windows.
+        os.mkdir = mkdir  # type: ignore
```

Comment:
why do we need to ignore the type?

Reply:
Because our mkdir function does not have exactly the same type signature. And fixing that requires unpleasant tricks to make it work both on python 2.7 and python 3.x, that I do not think it worth the complexity for this specific internal call to mkdir.